controllers/datadogmonitor/datadogmonitor_controller.go

```diff
@@ -0,0 +1,385 @@
+}
+
+func (r *Reconciler) create(logger logr.Logger, datadogMonitor *datadoghqv1alpha1.DatadogMonitor, status *datadoghqv1alpha1.DatadogMonitorStatus, now metav1.Time) error {
+	// Validate monitor in Datadog
+	err := r.validateMonitor(datadogMonitor)
+	if err != nil {
+		return err
+	}
+
+	// Create monitor in Datadog
+	m, err := r.createMonitor(datadogMonitor)
+	if err != nil {
+		return err
+	}
+	event := buildEventInfo(datadogMonitor.Name, datadogMonitor.Namespace, datadog.CreationEvent)
+	r.recordEvent(datadogMonitor, event)
+
+	// As this is a new monitor, add static information to status
+	status.ID = int(m.GetId())
+	creator := m.GetCreator()
+	status.Creator = creator.GetEmail()
+	createdTime := metav1.NewTime(m.GetCreated())
+	status.Created = &createdTime
+	status.Primary = true
+
+	// Set Created Condition
+	condition.UpdateDatadogMonitorConditions(status, now, datadoghqv1alpha1.DatadogMonitorConditionTypeCreated, corev1.ConditionTrue, "DatadogMonitor Created")
+	logger.Info("Created a new DatadogMonitor", "datadogMonitor.Namespace", datadogMonitor.Namespace, "datadogMonitor.Name", datadogMonitor.Name, "m.ID", m.GetId())
+
+	return nil
+}
+
+func (r *Reconciler) update(logger logr.Logger, datadogMonitor *datadoghqv1alpha1.DatadogMonitor, status *datadoghqv1alpha1.DatadogMonitorStatus, now metav1.Time) error {
+	// Validate monitor in Datadog
+	err := r.validateMonitor(datadogMonitor)
+	if err != nil {
+		return err
+	}
+
+	// Update monitor in Datadog
+	_, err = r.updateMonitor(datadogMonitor)
+	if err != nil {
+		return err
+	}
+
+	// Set Updated Condition
+	condition.UpdateDatadogMonitorConditions(status, now, datadoghqv1alpha1.DatadogMonitorConditionTypeUpdated, corev1.ConditionTrue, "DatadogMonitor Updated")
+	logger.Info("Updated DatadogMonitor", "DatadogMonitor.Namespace", datadogMonitor.Namespace, "datadogMonitor.Name", datadogMonitor.Name, "datadogMonitor.Status.ID", datadogMonitor.Status.ID)
+
+	return nil
+}
+
+func (r *Reconciler) get(logger logr.Logger, datadogMonitor *datadoghqv1alpha1.DatadogMonitor, newStatus *datadoghqv1alpha1.DatadogMonitorStatus) error {
+	// Get monitor from Datadog and update resource status if needed
+	m, err := r.getMonitor(datadogMonitor.Status.ID)
+	if err != nil {
+		return err
+	}
+
+	convertStateToStatus(m, newStatus)
+	logger.Info("Synced DatadogMonitor state", "DatadogMonitor.Namespace", datadogMonitor.Namespace, "datadogMonitor.Name", datadogMonitor.Name)
+
+	return nil
+}
+
+func (r *Reconciler) updateStatusIfNeeded(logger logr.Logger, datadogMonitor *datadoghqv1alpha1.DatadogMonitor, now metav1.Time, status *datadoghqv1alpha1.DatadogMonitorStatus, currentErr error, result ctrl.Result) (ctrl.Result, error) {
+	// Update Error and Active conditions
+	setErrorActiveConditions(status, now, currentErr)
+
+	if !apiequality.Semantic.DeepEqual(&datadogMonitor.Status, status) {
+		datadogMonitorCopy := datadogMonitor.DeepCopy()
```

Comment:
I think a datadogMonitor with `metadata` and `status` is enough as you update a sub-resource)